### PR TITLE
Accept actual type instead of string representation in Argument class

### PR DESCRIPTION
**Pull request by RobbeSneyders**
Needed some refactoring while working on #751.

Submitting this as a separate PR for easy reviewing.

### CODE REVIEW

File: src/fondant/core/component_spec.py [hunk 1]
```diff
@@ -208,7 +197,7 @@ def args(self) -> t.Mapping[str, Argument]:
                 name: Argument(
                     name=name,
                     description=arg_info["description"],
-                    type=arg_info["type"],
+                    type=pydoc.locate(arg_info["type"]),  # type: ignore
```
Comment (GeorgesLorre) [inline src/fondant/core/component_spec.py:200]: Is it normal I find no docs on pydoc.locate ?

**Review by GeorgesLorre — APPROVED** [review]

**Comment by GeorgesLorre** [comment]
No tests ? 🤔

File: src/fondant/core/component_spec.py (hunk 1 above)
Comment (RobbeSneyders) [reply src/fondant/core/component_spec.py:200]: Doesn't seem well documented indeed. Got it from [here](https://stackoverflow.com/a/29831586/4098821).

**Comment by RobbeSneyders** [comment]
> No tests ? 🤔

I didn't change any behavior. This code is covered indirectly.

**Pull request merged**